protocol.md

```diff
@@ -162,6 +163,67 @@ Receive Firmware Name and Version (after query)
 N  END_SYSEX         (0xF7)
 ```
 
+Report supported Firmata features
+---
+
+Report which optional Firmata features are supported by the firmware and the version of each feature. 
+
+As described in the [Feature registry documentation](https://github.com/firmata/protocol/blob/master/feature-registry.md),
+Firmata features are split into 2 categories: *Core features* and *Optional features*. The feature
+query reports the optional features that are supported by the firmware as well as the version
+of each supported optional feature.
+
+The feature query reduces the need to define a new pin mode for every introduced feature. Going
+forward pin modes will only be defined and maintained for microcontroller specific features such as 
+digital and analog I/O, PWM, I2C, UART, SPI, DAC, etc. The use of pin modes for non microcontroller
```

Comment:
> I'm not exactly sure what you mean here by "drill down" and "full introspection". Could you elaborate further?

This would require a multi-pass approach...

--- FIRST PASS ---
client: [describe board]
host: [board description] [gpio] [analog] [i2c] [spi]

--- SECOND PASS ---
client: [describe spi]
host: [spi description] [ss] [10] [mosi] [11] [miso] [12] [sclk] [13]

And so forth, and so on for each "feature". Providing full-introspection by allowing you to drill-down.